graphics/macgui/macwindowmanager.cpp

```diff
@@ -214,12 +214,8 @@ MacWindowManager::MacWindowManager(uint32 mode, MacPatterns *patterns, Common::L
 	if (g_system->getScreenFormat().isCLUT8())
 		g_system->getPaletteManager()->setPalette(palette, 0, ARRAYSIZE(palette) / 3);
 
-	_paletteSize = ARRAYSIZE(palette) / 3;
-	if (_paletteSize) {
-		_palette = (byte *)malloc(_paletteSize * 3);
-		memcpy(_palette, palette, _paletteSize * 3);
-	}
-	_paletteLookup.setPalette(_palette, _paletteSize);
+	_palette.set(palette, 0, ARRAYSIZE(palette) / 3);
+	_paletteLookup.setPalette(_palette);
```

Comment:
Is it possible to have only one copy of the palette data?

Reply:
PaletteLookup could have a const reference accessor for it's palette.